app/src/main/java/com/github/mik629/android/fundamentals/ui/moviedetails/FragmentMovieDetails.kt

```diff
@@ -1,90 +1,104 @@
 package com.github.mik629.android.fundamentals.ui.moviedetails
 
+import android.content.Context
 import android.os.Bundle
-import android.view.LayoutInflater
 import android.view.View
-import android.view.ViewGroup
 import androidx.fragment.app.Fragment
-import com.bumptech.glide.load.resource.drawable.DrawableTransitionOptions
+import androidx.fragment.app.viewModels
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.ViewModelProvider
+import by.kirich1409.viewbindingdelegate.viewBinding
 import com.github.mik629.android.fundamentals.BuildConfig
-import com.github.mik629.android.fundamentals.GlideApp
 import com.github.mik629.android.fundamentals.R
 import com.github.mik629.android.fundamentals.databinding.FragmentMovieDetailsBinding
-import com.github.mik629.android.fundamentals.domain.model.MovieItem
+import com.github.mik629.android.fundamentals.di.AppModule
+import com.github.mik629.android.fundamentals.di.buildGlideRequest
 import com.github.mik629.android.fundamentals.ui.global.ActorItemAdapter
+import com.github.mik629.android.fundamentals.ui.utils.setRating
+import com.google.android.material.snackbar.BaseTransientBottomBar
+import com.google.android.material.snackbar.Snackbar
 
-class FragmentMovieDetails : Fragment() {
-    private lateinit var binding: FragmentMovieDetailsBinding
+class FragmentMovieDetails : Fragment(R.layout.fragment_movie_details) {
+    private val binding by viewBinding(FragmentMovieDetailsBinding::bind)
 
     private val glideRequest by lazy {
-        GlideApp.with(this)
-            .asDrawable()
-            .thumbnail(0.1f)
-            .placeholder(R.drawable.ic_image_loading)
-            .fallback(R.drawable.ic_broken_image)
-            .transition(DrawableTransitionOptions.withCrossFade())
-            .fitCenter()
+        buildGlideRequest(this)
     }
+    private val actorItemAdapter by lazy {
+        ActorItemAdapter(glideRequest)
+    }
+
+    private var movieId: Long = 0
```

Comment:
remove this id and use arguments?.getLong(ARG_MOVIE_ID) ?: 0 exact in factoryProducer

Reply:
you mean by making factoryProducer an inner class? or how else it would get access to fragment's arguments 🤔

Reply:
private val viewModel: MovieDetailsViewModel by viewModels(
         factoryProducer = {
             val movieId: Long = (arguments?.getLong(ARG_MOVIE_ID) ?: 0)
             MovieDetailsViewModelFactory(movieId)
         }
     )